src/features/todos.ts

```diff
@@ -1,9 +1,61 @@
+import type { RootState } from '../app/store';
 import { Todo } from '../types/Todo';
 
-export const actions = {};
+export enum TodosActionType {
+  SetTodos = 'todos/SET',
+}
 
-const todosReducer = (): Todo[] => {
-  return [];
+type SetTodosAction = {
+  type: TodosActionType.SetTodos,
+  payload: Todo[],
+};
+
+const setTodos = (todos: Todo[]): SetTodosAction => ({
+  type: TodosActionType.SetTodos,
+  payload: todos,
+});
+
+export const TODOS_ACTIONS = { setTodos };
+
+export const TODOS_SELECTORS = {
+  getTodos: (state: RootState) => state.todos,
+  getPreparedTodos: (state: RootState) => {
+    const { todos, filter } = state;
+    const { query, status } = filter;
+
+    let preparedTodos: Todo[];
+
+    switch (status) {
+      case 'completed':
+        preparedTodos = [...todos].filter(todo => todo.completed);
```

Comment:
You don't need to create a new array with spread operator because `filter` method always returns a new array.